App/Services/UserNotifier.cs

```diff
@@ -90,20 +108,32 @@ public void HandleNotificationActivation(IDictionary<string, string> args)
 
         if (!Handlers.TryGetValue(handlerName, out var handler))
         {
-            logger.LogWarning("no action handler '{HandlerName}' found for notification activation, ignoring", handlerName);
+            _logger.LogWarning("no action handler '{HandlerName}' found for notification activation, ignoring", handlerName);
             return;
         }
 
-        dispatcherQueueManager.RunInUiThread(() =>
+        _dispatcherQueueManager.RunInUiThread(() =>
         {
             try
             {
                 handler.HandleNotificationActivation(args);
             }
             catch (Exception ex)
             {
-                logger.LogWarning(ex, "could not handle activation for notification with handler '{HandlerName}", handlerName);
+                _logger.LogWarning(ex, "could not handle activation for notification with handler '{HandlerName}", handlerName);
             }
         });
     }
 }
+
+public class DefaultNotificationHandler : INotificationHandler
```

Comment:
Can you make App implement this perhaps? Then it could just get registered like IDispatcherQueueManager does